vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/bandits/epsilon_greedy_bandit_processor.cc

```diff
@@ -25,13 +26,13 @@ namespace {
 constexpr double kDefaultArmValue = 1.0;
 constexpr int kDefaultArmPulls = 0;
 
-targeting::EpsilonGreedyBanditArmMap MaybeAddOrResetArms(
-    const targeting::EpsilonGreedyBanditArmMap& arms) {
-  targeting::EpsilonGreedyBanditArmMap updated_arms = arms;
+void MaybeAddOrResetArms(targeting::EpsilonGreedyBanditArmMap* arms) {
+  DCHECK(arms);
 
-  for (const char* const segment : targeting::GetSegments()) {
-    const auto iter = updated_arms.find(segment);
-    if (iter != updated_arms.cend()) {
+  for (const base::StringPiece value : targeting::GetSegments()) {
```

Comment:
`const base::StringPiece segment`?

Reply:
I think that `segment` is a better naming for std::string variable.